src/Microsoft.Performance.SDK/Processing/ProcessingSource.cs

```diff
@@ -46,8 +45,30 @@ protected ProcessingSource()
             // so unfortunately we have to use an older idiom of every constructor calling an
             // init method so that we can pass the concrete type's assembly.
             //
-            this.tableAssemblyProvider = () => this.GetType().Assembly;
-            this.additionalTablesProvider = () => new Dictionary<TableDescriptor, Action<ITableBuilder, IDataExtensionRetrieval>>();
+
+            this.tableDiscoverer = TableDiscovery.CreateForAssembly(this.GetType().Assembly);
```

Comment:
this ctor can just call `this(TableDiscovery.CreateForAssembly(this.GetType().Assembly)`, no?

Reply:
you can't call this in this() or base(). I documented this in the other constructor, I can copy that documentation here as well.